book/bayesian_optimization.qmd

```diff
@@ -0,0 +1,844 @@
+---
+
+# Bayesian Optimization {#sec-bayesian-optimization}
+
+```{r}
+set.seed(1005)
+```
+
+{{< include _setup.qmd >}}
+
+Black box optimization considers the optimization of a so-called black box function whose structure and analytical description is unknown, unexploitable or non-existent.
+As a result, the only observable information is the output value of the function given an input value.
+Usually, the objective function is subject so some box constraint, i.e., in the case of numeric variables the domain is assumed to be a hyperrectangle.
+Examples for black box optimization problems range from real-life experiments, e.g., crash tests or chemical reactions to computer simulations.
+
+Hyperparameter optimization (HPO, see @sec-optimization) is a prime example for black box optimization.
+Recall that in HPO, we configure a learner with a hyperparameter configuration and evaluate the learner via a resampling technique to measure its performance with the goal to find the optimal hyperparameter configuration.
+In general no analytical description for this mapping from a hyperparameter configuration to performance exists and gradient information is not available.
+Besides, evaluating the performance of a learner can take a substantial amount of time, making HPO an expensive black box optimization problem.
+
+Many algorithm classes exists that can be used for black box optimization that differ in how they tackle this optimization problem.
+In general, most black box optimizers work iterative, i.e., they propose new points for evaluation by making use of the information collected during the evaluation of previous points.
+Evolutionary strategies for example maintain a so-called population and generate new points for evaluation by choosing parents from that population and performing recombination operators such as mutation and crossover to generate the offspring of the next generation.
+In general, evolutionary strategies are very suited for black box optimization, however, if the overhead of the evaluation of the black box function becomes large (e.g., as in HPO), sample efficiency of an optimizer becomes highly relevant.
+Bayesian optimization (BO) describes a class of sample efficient iterative global black box optimization algorithms that make use of a so-called surrogate trained on observed data modelling the black box function.
+Based on this (non-linear) regression model and a so-called acquisition function quantifying the attractiveness of each point in the search space, the next candidate point is obtained that is to be evaluated next.
+After evaluating this candidate, the surrogate model is updated or re-trained and the process repeats itself until a termination criteria is met.
+Often, using BO results in very good optimization performance, especially if the overhead of the black box evaluation becomes large and optimization budget is tight.
```

Comment:
I think it needs to be clearer why we are including this chapter before going into some complex terminology. Personally this is an area I'm very interested in learning more about, but this intro makes it seem complicated and doesn't tell me in simple terms *why* I should care